### ISSUE TITLE
about loss function

### ISSUE THREAD

**Issue by lzy119** [poojahira/image-captioning-bottom-up-top-down#16]
hi pooja!
 i am reading your code,i have a question about loss function.
what does the 'loss_d' mean? There seems to be no introduction to it in the paper.
I am anticipating your reply. 
thank you！
`targets_d = torch.zeros(scores_d.size(0),scores_d.size(1)).to(device)
            targets_d.fill_(-1)

            for length in decode_lengths:
                targets_d[:,:length-1] = targets[:,:length-1]

            # Remove timesteps that we didn't decode at, or are pads
            # pack_padded_sequence is an easy trick to do this
            scores_copy = scores.clone()
            scores, _ = pack_padded_sequence(scores, decode_lengths, batch_first=True)
            targets, _ = pack_padded_sequence(targets, decode_lengths, batch_first=True)

            # Calculate loss
            loss_d = criterion_dis(scores_d,targets_d.long())
            loss_g = criterion_ce(scores, targets)
            loss = loss_g + (10 * loss_d)`

**Comment by poojahira** [comment]
Hi, it was just an extra loss function I was experimenting with for my project. You can safely exclude it from the code for your purposes.

**Comment by lzy119** [comment]
thank you for your reply. i get it. Do you think it worked?

**Comment by lzy119** [comment]
@poojahira

**Comment by poojahira** [comment]
Yes it did if I recall correctly.

**Comment by lzy119** [comment]
thank you  for your reply. i am trying to train this model by 2 GPUs. However, the GPU memory usage is extremely imbalanced. I can understand that one GPU is set to gather and store all outputs. I wonder if there is any way I can balance the memory usage?  So i attempt to split the loss among gpus. 
but it syill hasn't changed,the gpu memory usage is imbalanced. Do you have any suggestions?
my GPU memory is 12G.

**Comment by lzy119** [comment]
hi ,i found that i attempt to split the loss among gpus. this method is useful. but i can't understand why gpu memory keeps increasing? Do you know why?

**Comment by lzy119** [comment]
@poojahira

**Comment by lzy119** [comment]
i found the reason! this is because losses accumulate as training progresses. thank you for your support!

**Comment by poojahira** [comment]
Sure, glad you found out why. Helps me as well. Thanks and good luck.
